metaflow/plugins/argo/argo_workflows.py

```diff
@@ -247,24 +296,145 @@ def _process_parameters(self):
+                parameters = {}
+                # TODO: Add a check to guard against names starting with numerals(?)
+                if not re.match(r"^[A-Za-z0-9_.-]+$", event["name"]):
+                    raise ArgoWorkflowsException(
+                        "Invalid event name *%s* in *@trigger* decorator. Only "
+                        "alphanumeric characters, underscores(_), dashes(-) and "
+                        "dots(.) are allowed." % event["name"]
+                    )
+                for key, value in event.get("parameters", {}).items():
+                    if not re.match(r"^[A-Za-z0-9_]+$", value):
+                        raise ArgoWorkflowsException(
+                            "Invalid event payload key *%s* for event *%s* in "
+                            "*@trigger* decorator. Only alphanumeric characters and "
+                            "underscores(_) are allowed." % (value, event["name"])
+                        )
+                    if key.lower() not in params:
+                        raise ArgoWorkflowsException(
+                            "Parameter *%s* defined in the event mappings for "
+                            "*@trigger* decorator not found in the flow." % key
+                        )
+                    if key.lower() in seen:
+                        raise ArgoWorkflowsException(
+                            "Duplicate entries for parameter *%s* defined in the "
+                            "event mappings for *@trigger* decorator." % key.lower()
+                        )
+                    seen.add(key.lower())
+                    parameters[key.lower()] = value
+                event["parameters"] = parameters
+                event["type"] = "event"
+            triggers.extend(self.flow._flow_decorators.get("trigger")[0].triggers)
+
+        # @trigger_on_finish decorator
+        if self.flow._flow_decorators.get("trigger_on_finish"):
+            for event in self.flow._flow_decorators.get("trigger_on_finish")[
+                0
+            ].triggers:
+                # TODO: Also sanity check project and branch names
+                if not re.match(r"^[A-Za-z0-9_]+$", event["flow"]):
+                    raise ArgoWorkflowsException(
+                        "Invalid flow name *%s* in *@trigger_on_finish* "
+                        "decorator. Only alphanumeric characters and "
+                        "underscores(_) are allowed." % (value, event["flow"])
+                    )
+                # Throw an error if we are unable to impute project for specified
+                # branch.
+                if (
+                    not event.get("project")
+                    and not current.get("project_name")
+                    and event.get("branch")
+                ):
+                    raise ArgoWorkflowsException(
+                        "No project specified for branch in *@trigger_on_finish* "
```

Comment:
Another case is that project can be specified even though branch is left empty. This will listen to `project/None` for completion. Is this a valid use-case or should there be a check that only both values are available at once?